src/basic/virt.c

```diff
@@ -472,8 +463,18 @@ int detect_vm(void) {
                 return r;
         if (r == VIRTUALIZATION_VM_OTHER)
                 other = true;
-        else if (r != VIRTUALIZATION_NONE)
-                goto finish;
+        else if (r != VIRTUALIZATION_NONE) {
+                 /* If we are Dom0, then we expect to not report as a VM. However, as we might be nested
+                  * inside another hypervisor which can be detected via the CPUID check, wait to report this
+                  * until after the CPUID check. */
+                xen_dom0 = detect_vm_xen_dom0();
+                if (xen_dom0 < 0)
+                        return xen_dom0;
+                else if (xen_dom0 > 0)
+                        r = VIRTUALIZATION_NONE;
+                else
+                        goto finish;
```

Comment:
please write this like this:

```c
if (xen_dom0 < 0)
        return xen_dom0;
if (xen_dom0 == 0)
        goto finish;

r = VIRTUALIZATION_NONE;
…
```

i.e. nest the jumps (i.e. the return + goto) in the conditional blocks, and thus get rid of needless indentation of the main codepath here. i.e. less indentation is a good thing

Reply:
Done